integrations/opennms-wsman-asset-provisioning-adapter/src/main/java/org/opennms/netmgt/provision/WsManAssetProvisioningAdapter.java

```diff
@@ -0,0 +1,463 @@
+import org.opennms.netmgt.dao.WSManConfigDao;
+import org.opennms.netmgt.dao.api.NodeDao;
+import org.opennms.netmgt.events.api.EventConstants;
+import org.opennms.netmgt.events.api.EventForwarder;
+import org.opennms.netmgt.events.api.annotations.EventHandler;
+import org.opennms.netmgt.events.api.annotations.EventListener;
+import org.opennms.netmgt.model.OnmsAssetRecord;
+import org.opennms.netmgt.model.OnmsIpInterface;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.netmgt.model.events.EventBuilder;
+import org.opennms.netmgt.model.events.snmp.SyntaxToEvent;
+
+import org.opennms.netmgt.xml.event.Event;
+import org.opennms.netmgt.xml.event.Parm;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeanWrapper;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.PropertyAccessorFactory;
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.TransactionCallback;
+import org.springframework.util.Assert;
+
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import com.google.common.collect.Lists;
+
+@EventListener(name="WsManAssetProvisioningAdapter")
+public class WsManAssetProvisioningAdapter extends SimplerQueuedProvisioningAdapter {
+    private static final Logger LOG = LoggerFactory.getLogger(WsManAssetProvisioningAdapter.class);
+
+        private NodeDao m_nodeDao;
+        private EventForwarder m_eventForwarder;
+        private WsManAssetAdapterConfig m_config;
+        private WSManClientFactory m_factory = new CXFWSManClientFactory();
+       // private WSManClientFactory m_factory;
+        private WSManConfigDao m_wsManConfigDao;
+
+        /**
+         * Constant <code>NAME="WsManAssetProvisioningAdapter"</code>
+         */
+        public static final String NAME = "WsManAssetProvisioningAdapter";
+
+        public WsManAssetProvisioningAdapter() {
+                super(NAME);
+
+                // Set the default time delay to 30 seconds
+                this.setDelay(30);
+                this.setTimeUnit(TimeUnit.SECONDS);
+        }
+
+        /**
+         * Creating a custom schedule for this adapter.  We need to make sure that the node has a system object ID set
+         * and that it has had enough time for that to have happened.
+         * @param nodeId
+         * @param adapterOperationType
+         * @return
+         */
+        @Override
+        AdapterOperationSchedule createScheduleForNode(int nodeId, AdapterOperationType adapterOperationType) {
+                AdapterOperationSchedule aos = new AdapterOperationSchedule(m_delay, 60, 3, m_timeUnit);
+                LOG.info("createScheduleForNode: Scheduling {} for nodeid {} with schedule: {}", adapterOperationType, nodeId, aos);
+                return aos;
+        }
+
+        @Override
+        public boolean isNodeReady(AdapterOperation op) {
+                return true;
+        }
+
+        /**
```

Comment:
Delete javadoc lines 128 to 134. Better to have no javadoc on the method than incorrect/incomplete.